Makefile

```diff
@@ -63,55 +51,22 @@ questasim-sim-opt: questasim-sim
 vcs-sim:
 	fusesoc --cores-root . run --no-export --target=sim --tool=vcs $(FUSESOC_FLAGS) --setup --build eslepfl:systems:cgra-x-heep 2>&1 | tee buildsim.log
 
-run-helloworld-verilator: mcu-gen verilator-sim app-helloworld
-	cd ./build/eslepfl_systems_cgra-x-heep_0/sim-verilator; \
-	./Vtestharness +firmware=../../../sw/x_heep_applications/hello_world/hello_world.hex; \
-	cat uart0.log; \
-	cd ../../..;
 
-run-cgra-test-verilator: mcu-gen verilator-sim app-cgra-test
+## Generates the build output for a given application
+## Uses verilator to simulate the HW model and run the FW
+## UART Dumping in uart0.log to show recollected results
+run-verilator: 
+	$(MAKE) app PROJECT=$(PROJECT)
 	cd ./build/eslepfl_systems_cgra-x-heep_0/sim-verilator; \
-	./Vtestharness +firmware=../../../sw/applications/cgra_func_test/main.hex; \
-	cat uart0.log; \
-	cd ../../..;
-
-run-helloworld-questasim: mcu-gen questasim-sim app-helloworld
-	cd ./build/eslepfl_systems_cgra-x-heep_0/sim-modelsim; \
-	make run PLUSARGS="c firmware=../../../sw/x_heep_applications/hello_world/hello_world.hex"; \
+	./Vtestharness +firmware=../../../sw/build/main.hex; \
 	cat uart0.log; \
 	cd ../../..;
 
-run-cgra-test-questasim: mcu-gen questasim-sim app-cgra-test
-	cd ./build/eslepfl_systems_cgra-x-heep_0/sim-modelsim; \
-	make run PLUSARGS="c firmware=../../../sw/applications/cgra_func_test/main.hex"; \
-	cat uart0.log; \
-	cd ../../..;
+run-fpga:
+	$(MAKE) app PROJECT=$(PROJECT) LINKER=flash_load TARGET=pynq-z2
+	( cd hw/vendor/esl_epfl_x_heep/sw/vendor/yosyshq_icestorm/iceprog && make clean && make all ) ;\
+	sudo $(MAKE) flash-prog ;\
+	sudo picocom -b 115200 -r -l --imap lfcrlf /dev/ttyUSB2
```

Comment:
I would add that part to the readme so people can launch the make run-fpga even without picocom and they can use their favorite serial monitoring tool. Or use a tool that is present by default (maybe screen is part of Ubuntu by default).

Reply:
You mean add it as a step on the readme, but not on the Makefile? We could keep two in the Makefile maybe? With and without picocom